lib/assets/javascripts/chapman_stories_feed.js

```diff
@@ -0,0 +1,191 @@
+  function Plugin(element, options) {
+    this.element = element;
+    this.$element = $(element);
+
+    // jQuery has an extend method which merges the contents of two or
+    // more objects, storing the result in the first object. The first object
+    // is generally empty as we don't want to alter the default options for
+    // future instances of the plugin
+    this.settings = $.extend({}, defaults, options);
+    this._defaults = defaults;
+    this._name = pluginName;
+    this.init();
+  }
+
+  // Avoid Plugin.prototype conflicts
+  $.extend(Plugin.prototype, {
+    init: function() {
+      // Place initialization logic here. You already have access to the DOM element and
+      // the options via the instance, e.g. this.element and this.settings.
+
+      // Data attribute values
+      this.feedUrl          = this.parseFeedUrl();
+      this.topStoryUrl      = this.$element.data('chapmanStoriesTopStoryUrl');
+      this.title            = this.$element.data('chapmanStoriesTitle') || this.settings.dataTitle;
+      this.backgroundColor  = this.$element.data('chapmanStoriesBgColor') || this.settings.dataBgColor;
+      this.buttonLabel      = this.$element.data('chapmanStoriesButtonLabel') || this.settings.dataButtonLabel;
+      this.buttonUrl        = this.$element.data('chapmanStoriesButtonUrl') || this.settings.dataButtonUrl;
+      this.buildOutWidget();
+    },
+
+    //this is a temporary fix to remove trailing slash from blog site urls(that is how they are stored on inside db)
+    //Ideally, inside query should be able to match urls with or without trailing slash
+    parseFeedUrl: function() {
+      var parsedUrl = this.$element.data('chapmanStoriesFeedUrl') || this.settings.dataFeedUrl;
+      parsedUrl = parsedUrl.replace(/\/?$/, '');
+      return parsedUrl;
+    },
+
+    buildOutWidget: function() {
+      var $outerContainer = this.buildOuterContainer();
+      var $topStory = this.buildTopStoryElement();
+      var $storiesFeed = this.buildStoriesFeedElement();
+
+      if ( this.backgroundColor ) {
+        this.$element.css('backgroundColor', this.backgroundColor);
+      }
+
+      this.$element.html($outerContainer);
+      $outerContainer.append($topStory);
+      $outerContainer.append($storiesFeed);
+    },
+
+    buildOuterContainer: function() {
+      var $outerContainer = $('<div />').addClass('outer-container');
+      var $title = $('<h2 />').addClass('social-feed-title').text(this.title);
+
+      $outerContainer.append($title);
+      return $outerContainer;
+    },
+
+    buildTopStoryElement: function() {
+      var classes = 'feed-column top-story';
+      var loadingHtml = '<p>Loading top story...</p>'
+      var $topStoryElement = $('<div />').addClass(classes).html(loadingHtml);
+      
+      //if top story post is not specified, default to the top story from the specified blog feed
+      var params = this.topStoryUrl ? { url: this.topStoryUrl } : { source: this.feedUrl };
```

Comment:
Send both and let the backend (Inside) figure it out. That way, if we decide we want to change the logic or output, we can do it there and not worry about coordinating with this.